lib/Importer/ONNX.cpp

```diff
@@ -243,29 +243,58 @@ bool ONNXModelLoader::loadOperator(const ONNX_NAMESPACE::NodeProto &op) {
     // Pads : {pad_top, pad_left, pad_bottom, pad_right}
     std::vector<unsigned_t> pads = getPads(dict);
 
+    // Load the inputs:
     auto in = getNodeValueOrCreateVariableByName(op.input(0));
-    Tensor *w = getTensorByName(op.input(1));
-
-    // Transpose the weights to the right format. Glow expects to read the
-    // weights in the format CRSK. ONNX stores the operators as KCRS.
-    // C - output_depth, R - filter_height, S - filter_width, K - input_depth.
-    Tensor wtag;
-    w->transpose(&wtag, NCHW2NHWC);
+    Node *filterTransposeNode = NULL;
+    std::vector<unsigned_t> kernelShape(2);
+    size_t depth;
+
+    if (hasTensorWithName(op.input(1))) {
+      // The filter is a constant tensor.
+      Tensor *w = getTensorByName(op.input(1));
+
+      // Transpose the weights to the right format. Glow expects to read the
+      // weights in the format CRSK. ONNX stores the operators as KCRS.
+      // C - output_depth, R - filter_height, S - filter_width, K - input_depth.
+      Tensor wtag;
+      w->transpose(&wtag, NCHW2NHWC);
+
+      // The structure of the conv weigts is: NHWC. We take the C, which is the
+      // number of filters. We use this value to calculate the size of the bias
+      // if it is not specified.
+      depth = wtag.dims()[0];
+
+      // Construct the Filter field.
+      auto *filterVariable =
+          G_.getParent()->createVariable("conv.filter", wtag);
+      filterTransposeNode = filterVariable;
+
+      // Get the kernel shape from the input.
+      kernelShape[0] = filterVariable->dims()[1];
+      kernelShape[1] = filterVariable->dims()[2];
+    } else {
+      // The filter is generated by a node (example with depthwise
+      // convolution). Transpose it to meet the Glow expectation.
+      auto filter_node = getNodeValueOrCreateVariableByName(op.input(1));
+      filterTransposeNode = G_.createTranspose(opName, filter_node, NCHW2NHWC);
 
-    // The structure of the conv weigts is: NHWC. We take the C, which is the
-    // number of filters. We use this value to calculate the size of the bias
-    // if it is not specified.
-    size_t depth = wtag.dims()[0];
+      const NodeValue filterValue = filterTransposeNode->getNthResult(0);
+      depth = filterValue.dims()[0];
 
-    // Construct the Filter field.
-    auto *filter = G_.getParent()->createVariable("conv.filter", wtag);
+      // Get the kernel shape from the input.
+      kernelShape[0] = filterValue.dims()[1];
+      kernelShape[1] = filterValue.dims()[2];
+    }
 
-    std::vector<unsigned_t> kernels(2);
+    // Extra check when the 'kernel_shape' attribute exists.
+    // The 'kernel_shape' attribute is redundant not mandatory.
     if (dict.count("kernel_shape")) {
-      kernels = getShape<unsigned_t>(dict.at("kernel_shape"));
-    } else {
-      kernels[0] = filter->dims()[1];
-      kernels[1] = filter->dims()[2];
+      std::vector<unsigned_t> kernelShapeAttribute(2);
```

Comment:
nit: Effect of initialization with `(2)` gets immediately overridden by following assignment. You can do simply:
`auto kernelShapeAttribute = getShape<unsigned_t>(dict.at("kernel_shape"));`
Note, that when compiling in release, asserts are eliminated. So `kernelShapeAttribute` will be considered unused and we'll see compiler warning. To defeat this, we add `(void)kernelShapeAttribute;`

Reply:
Thanks for the feedback. I preferred the explicit typing and used `std::vector<unsigned_t> kernelShapeAttribute = getShape<unsigned_t>(dict.at("kernel_shape"));`. I also added the suggested statement to avoid compilation warnings in release mode.